### PR TITLE
Add better documentation for dict type arguments

### PR DESCRIPTION
**Pull request by gabrielfortin**

### CODE REVIEW

File: library/f5bigip_gtm_listener.py [hunk 1]
```diff
@@ -104,6 +104,15 @@
         description:
             - Specifies the type of source address translation enabled for the listener as well as the pool that the
               source address translation will use.
+        suboptions:
+            pool: 
+                description:
+                    - Specifies the name of a SNAT pool used by the specified listener.
+
```
Comment (erjac77) [inline library/f5bigip_gtm_listener.py]: Remove extra line break

File: library/f5bigip_ltm_profile_dhcpv4.py [hunk 2]
```diff
@@ -37,6 +37,44 @@
...
+                suboptions:
+                    format:
+                        description:
+                            - Specifies the user-name format.
+                        choices: ['mac-address', 'mac-and-relay-id', 'tcl-snippet']
+                    suboption_id1:
+                        description:
+                            - The relay-agent option (option 82) first suboption ID.
+                        default: 1
+                    suboption_id2
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv4.py]: Miss a colon after `suboption_id2`

File: library/f5bigip_ltm_profile_dhcpv4.py [hunk 3]
```diff
@@ -37,6 +37,44 @@
...
+                        description:
+                            - The relay-agent option (option 82) second suboption ID.
+                        default: 2
+                    separator1
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv4.py]: Miss a colon after `separator1`

File: library/f5bigip_ltm_profile_dhcpv4.py [hunk 4]
```diff
@@ -37,6 +37,44 @@
...
+                        description:
+                            - A string that is used to concatenate the MAC address and the relay-agent info option (option 82)
+                              to create the authentication user-name.
+                        default: @
+                    separator2
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv4.py]: Miss a colon after `separator2`

File: library/f5bigip_ltm_profile_dhcpv4.py [hunk 5]
```diff
@@ -37,6 +37,44 @@
...
+                        description:
+                            - A string that is used to concatenate the relay-agent info option (option 82) suboptions 1 and 2 to
+                              create the authentication user-name.
+                        default: @
+                    tcl_snippet
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv4.py]: Miss a colon after `tcl_snippet`

File: library/f5bigip_ltm_profile_dhcpv4.py [hunk 6]
```diff
@@ -78,6 +146,41 @@
...
+                suboptions:
+                    format:
+                        description:
+                            - Specifies the user-name format.
+                        choices: ['mac-address', 'mac-and-relay-id', 'tcl-snippet']
+                    suboption_id1:
+                        description:
+                            - The relay-agent option (option 82) first suboption ID.
+                        default: 1
+                    suboption_id2
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv4.py]: Miss a colon after `suboption_id2`

File: library/f5bigip_ltm_profile_dhcpv4.py [hunk 7]
```diff
@@ -78,6 +146,41 @@
...
+                        description:
+                            - The relay-agent option (option 82) second suboption ID.
+                        default: 2
+                    separator1
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv4.py]: Miss a colon after `separator1`

File: library/f5bigip_ltm_profile_dhcpv4.py [hunk 8]
```diff
@@ -78,6 +146,41 @@
...
+                        description:
+                            - A string that is used to concatenate the MAC address and the relay-agent info option (option 82)
+                              to create the authentication user-name.
+                        default: @
+                    separator2
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv4.py]: Miss a colon after `separator2`

File: library/f5bigip_ltm_profile_dhcpv4.py [hunk 9]
```diff
@@ -78,6 +146,41 @@
...
+                        description:
+                            - A string that is used to concatenate the relay-agent info option (option 82) suboptions 1 and 2 to
+                              create the authentication user-name.
+                        default: @
+                    tcl_snippet
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv4.py]: Miss a colon after `tcl_snippet`

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 10]
```diff
@@ -37,6 +37,44 @@
...
+                suboptions:
+                    format:
+                        description:
+                            - Specifies the user-name format.
+                        choices: ['mac-address', 'mac-and-relay-id', 'tcl-snippet']
+                    suboption_id1:
+                        description:
+                            - The relay-agent option (option 82) first suboption ID.
+                        default: 1
+                    suboption_id2
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Miss a colon after `suboption_id2`

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 11]
```diff
@@ -37,6 +37,44 @@
...
+                        description:
+                            - The relay-agent option (option 82) second suboption ID.
+                        default: 2
+                    separator1
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Miss a colon after `separator1`

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 12]
```diff
@@ -37,6 +37,44 @@
...
+                        description:
+                            - A string that is used to concatenate the MAC address and the relay-agent info option (option 82)
+                              to create the authentication user-name.
+                        default: @
+                    separator2
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Miss a colon after `separator2`

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 13]
```diff
@@ -37,6 +37,44 @@
...
+                        description:
+                            - A string that is used to concatenate the relay-agent info option (option 82) suboptions 1 and 2 to
+                              create the authentication user-name.
+                        default: @
+                    tcl_snippet
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Miss a colon after `tcl_snippet`

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 14]
```diff
@@ -77,9 +133,59 @@
...
+                suboptions:
+                    format:
+                        description:
+                            - Specifies the user-name format.
+                        choices: ['mac-address', 'mac-and-relay-id', 'tcl-snippet']
+                    suboption_id1:
+                        description:
+                            - The relay-agent option (option 82) first suboption ID.
+                        default: 1
+                    suboption_id2
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Miss a colon after `suboption_id2`

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 15]
```diff
@@ -77,9 +133,59 @@
...
+                        description:
+                            - The relay-agent option (option 82) second suboption ID.
+                        default: 2
+                    separator1
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Miss a colon after `separator1`

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 16]
```diff
@@ -77,9 +133,59 @@
...
+                        description:
+                            - A string that is used to concatenate the MAC address and the relay-agent info option (option 82)
+                              to create the authentication user-name.
+                        default: @
+                    separator2
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Miss a colon after `separator2`

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 17]
```diff
@@ -77,9 +133,59 @@
...
+                        description:
+                            - A string that is used to concatenate the relay-agent info option (option 82) suboptions 1 and 2 to
+                              create the authentication user-name.
+                        default: @
+                    tcl_snippet
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Miss a colon after `tcl_snippet`

**Review by erjac77 — CHANGES_REQUESTED** [review]

**Comment by gabrielfortin** [comment]
Fixed

File: library/f5bigip_ltm_profile_dhcpv6.py [hunk 18]
```diff
@@ -69,6 +107,24 @@
...
+                    - Specifies if the user wants the DHCP relay agent to insert option 37 or not.
+                default: false
+                choices: ['true', 'false']
+            remove:
+                description:
+                    - Specifies if the user wants the DHCP relay agent to remove option 37 from the server-to-client traffic
+                      or not.
+                default: false
+                choices: ['true', 'false']
+            enterprise-number:
```
Comment (erjac77) [inline library/f5bigip_ltm_profile_dhcpv6.py]: Should be underscore

**Review by erjac77 — CHANGES_REQUESTED** [review]

**Pull request merged**